test/Log/LogServiceTest.php

```diff
@@ -291,6 +260,25 @@ public function testSetRunId()
         $this->assertEquals('customId', $logService->getRunId());
     }
 
+    /**
+     * Test that format's %runId% placeholder is being replaced by runId.
+     */
+    public function testPrepareFormat()
```

Comment:
Good tes. However we can still remove the call from `createLogger` and no test will break.

Reply:
Now that we do an additional test anyway, some additional stuff. What are we not testing here for this new function? What happens if the argument `$format === null`? Is that what we want to happen? Also does the function handle `PHP_EOL` as expected? Meaning does it deliver a string with `PHP_EOL` at the end and only one `PHP_EOL` at the end, even if we used an argument that already had `PHP_EOL` at the end.

Those additional test aren't terribly critical in this case, but it is an important part of testing to verify behavior if somethings isn't as it should be, bad arguments, bad configuration. That gets forgotten a lot.